src/ar_config.erl

```diff
@@ -0,0 +1,171 @@
+-module(ar_config).
+-export([parse/1]).
+-include("ar.hrl").
+-include("ar_config.hrl").
+
+%%%===================================================================
+%%% Public interface.
+%%%===================================================================
+
+parse(Config) when is_binary(Config) ->
+	case ar_serialize:json_decode(Config) of
+		{ok, JsonValue} -> parse_options(JsonValue);
+		{error, _} -> {error, bad_json, Config}
+	end.
+
+%%%===================================================================
+%%% Private functions.
+%%%===================================================================
+
+
+parse_options({KVPairs}) when is_list(KVPairs) ->
+	parse_options(KVPairs, #opts{});
+parse_options(JsonValue) ->
+	{error, root_not_object, JsonValue}.
+
+parse_options([{_, null} | Rest], Opts) ->
+	parse_options(Rest, Opts);
+parse_options([{<<"config_file">>, _} | _], _) ->
+	{error, config_file_set};
```

Comment:
I can imagine a user getting confused when creating a config file for the first time if he get the error "config_file_set". Can we explain this a little bit better? E.g. something like "config_file in the config file is not allowed"

Reply:
My thinking while writing this function was that it's caller should turn the atoms into user-friendly messages. Although we could do this now, if someone ran into an error as things are now, we could easily understand what's wrong.